File: src/main/java/com/github/frcsty/frozenactions/actions/broadcast/JsonBroadcastAction.java
package com.github.frcsty.frozenactions.actions.broadcast;

import com.github.frcsty.frozenactions.actions.Action;
import com.github.frcsty.frozenactions.actions.ActionContext;
import org.bukkit.Bukkit;
import org.jetbrains.annotations.NotNull;

public final class JsonBroadcastAction implements Action {

    @Override
    public @NotNull String getId() {
        return "JSONBROADCAST";
    }

    @Override
    public void run(final @NotNull ActionContext context) {
        Bukkit.getOnlinePlayers().forEach(player ->
            Bukkit.dispatchCommand(Bukkit.getConsoleSender(), "tellraw " + player.getName() + " " + context.getData())
        );
    }
}

File: src/main/kotlin/com/github/frcsty/frozenactions/actions/broadcast/JsonBroadcastAction.kt
package com.github.frcsty.frozenactions.actions.broadcast

import com.github.frcsty.frozenactions.actions.Action
import org.bukkit.Bukkit
import org.bukkit.entity.Player

object JsonBroadcastAction : Action {

    override val id = "JSONBROADCAST"

    override fun run(player: Player, data: String) {
        Bukkit.getServer().onlinePlayers.forEach { Bukkit.dispatchCommand(Bukkit.getConsoleSender(), "tellraw ${it.name} $data") }
    }
}